### PR TITLE
Fix: Azure endpoint return all users from ioet

### PR DESCRIPTION
**Pull request by cxcarvaj**

### CODE REVIEW

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
-const SlackClient = require("./slack_client");
-const { CLOCK_OUT_MESSAGE, CLOCK_OUT_MESSAGE_MIDNIGHT } = require("./constants");
+const _ = require('lodash');
+const CosmosClient = require('@azure/cosmos').CosmosClient;
+const config = require('./config');
+const TimeEntry = require('./time_entry');
+const MsalClient = require('./msal_client');
+const TimeEntryDao = require('./time_entry_dao');
+const SlackClient = require('./slack_client');
+const { CLOCK_OUT_MESSAGE, CLOCK_OUT_MESSAGE_MIDNIGHT } = require('./constants');
```
Comment (madinya) [inline nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:8]: Is it necesary to change between quote and single quote?

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-  context.log(
-    `I am going to check how many entries were not clocked out ${new Date()}`
-  );
+  context.log(`I am going to check how many entries were not clocked out ${new Date()}`);
 
   const { endpoint, key, databaseId } = config;
   const client = new CosmosClient({ endpoint, key });
   const database = client.database(databaseId);
-  const container = database.container("time_entry");
+  const container = database.container('time_entry');
```
Comment (madinya) [inline nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:16]: Is it necesary to change between quote and single quote?

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js [hunk 3]
```diff
@@ -30,7 +28,7 @@ const doClockOut = async (context) => {
   await Promise.all(
     entries.map(async (timeEntryAsJson) => {
       const timeEntry = new TimeEntry(timeEntryAsJson);
-      const { userName, userEmail } = findUserData(users, timeEntry.timeEntry.owner_id);
+      const { userName, userEmail } = findUserData( users, timeEntry.timeEntry.owner_id);
```
Comment (madinya) [inline nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:31]: can you try to remove the blank space? changes should be only what it was actually changed. Maybe it was an auto format?

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js [hunk 4]
```diff
@@ -40,11 +38,10 @@ const doClockOut = async (context) => {
         timeEntryAsJson.end_date = timeEntry.getTimeToClockOut();
         await container.item(timeEntryAsJson.id, timeEntryAsJson.tenant_id).replace(timeEntryAsJson);
         totalClockOutsExecuted++;
-      }
-
-      else if (timeEntry.needsToBeClockedOutMidnight()) {
+        
+      } else if (timeEntry.needsToBeClockedOutMidnight()) {
```
Comment (madinya) [inline nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:42]: if it was no changed can you keep the same format as before?

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js [hunk 5]
```diff
@@ -53,19 +50,17 @@ const doClockOut = async (context) => {
     })
   );
 
-  context.log(
-    `I just clocked out ${totalClockOutsExecuted} entries, thanks are not needed...`
-  );
+  context.log(`I just clocked out ${totalClockOutsExecuted} entries, thanks are not needed...`);
```
Comment (madinya) [inline nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:53]: Here looks better the format change

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js (hunk 1 above)
Comment (madinya) [reply nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:8]: If it's to keep a single quote standar on file it's okay also the following comments.

**Review by madinya — COMMENTED** [review]
Looks good to me, I left some comments to clarify prior the approve. Also it could be better if it can be checked by Sandro if it was not already reviewed.

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js (hunk 1 above)
Comment (cxcarvaj) [reply nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:8]: Yep, it's just to keep a standard.

File: nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js (hunk 3 above)
Comment (cxcarvaj) [reply nodejs-functions/src/handlers/automatic-clock-outs/clock_out.js:31]: Sure, right away!

**Review by madinya — APPROVED** [review]

**Review by heikergil — APPROVED** [review]

File: nodejs-functions/src/handlers/automatic-clock-outs/config.js [hunk 6]
```diff
@@ -8,8 +8,8 @@ const config = {
     clientSecret: process.env["CLIENT_SECRET"],
     slackApiToken: process.env["SLACK_TOKEN_NOTIFY"],
     userNameMS: process.env["USER_NAME_MS"],
-    userPasswordMS: process.env["USER_PASSWORD_MS"]
-    
+    userPasswordMS: process.env["USER_PASSWORD_MS"],
+    B2CLogin = process.env["B2C_LOGIN"]
```
Comment (scastillo-jp) [inline nodejs-functions/src/handlers/automatic-clock-outs/config.js:12]:
```suggestion
    b2cLogin = process.env["B2C_LOGIN"]
```

File: nodejs-functions/src/handlers/automatic-clock-outs/msal_client.js [hunk 7]
```diff
@@ -2,9 +2,8 @@ const axios = require('axios');
 const config = require('./config');
 
 const getToken = async () => {
-  const { clientId, userNameMS, userPasswordMS } = config;
-  const endpoint =
-    'https://ioetec.b2clogin.com/ioetec.onmicrosoft.com/B2C_1_accesstoken/oauth2/v2.0/token';
+  const { clientId, userNameMS, userPasswordMS, B2CLogin } = config;
```
Comment (scastillo-jp) [inline nodejs-functions/src/handlers/automatic-clock-outs/msal_client.js:5]:
```suggestion
  const { clientId, userNameMS, userPasswordMS, b2cLogin } = config;
```

File: nodejs-functions/src/handlers/automatic-clock-outs/msal_client.js [hunk 8]
```diff
@@ -2,9 +2,8 @@ const axios = require('axios');
...
+  const endpoint = B2CLogin;
```
Comment (scastillo-jp) [inline nodejs-functions/src/handlers/automatic-clock-outs/msal_client.js:6]:
```suggestion
  const endpoint = b2cLogin;
```

**Review by scastillo-jp — APPROVED** [review]

**Review by madinya — APPROVED** [review]
LGTM

**Pull request merged**